internal/beater/beater.go

```diff
@@ -379,12 +379,12 @@ func (s *Runner) Run(ctx context.Context) error {
 		finalBatchProcessor,
 	}
 
-	agentConfigFetcher, err := newAgentConfigFetcher(ctx, s.config, kibanaClient, newElasticsearchClient)
+	agentConfigFetcher, fetcherRunFunc, err := newAgentConfigFetcher(ctx, s.config, kibanaClient, newElasticsearchClient)
 	if err != nil {
 		return err
 	}
 	g.Go(func() error {
-		return agentConfigFetcher.Run(ctx)
+		return fetcherRunFunc(ctx)
 	})
```

Comment:
Should there be a fetcherRunFunc != nil check? Looks like newAgentConfigFetcher may return a nil func. Or perhaps it should always return a non-nil func in non error cases.

Reply:
My bad. It certainly needs a nil check